src/OpenLoco/src/Vehicles/VehicleManager.cpp

```diff
@@ -56,12 +57,63 @@ namespace OpenLoco::VehicleManager
         }
     }
 
+    // 0x004C39D4
+    static void determineAvailableVehicleTypes(Company& company)
+    {
+        uint16_t availableTypes = 0;
+
+        for (uint32_t i = 0; i < ObjectManager::getMaxObjects(ObjectType::vehicle); ++i)
+        {
+            if (!(company.unlockedVehicles[i & 0x1F] & 1ULL << (i / 32)))
+            {
+                continue;
+            }
+
+            auto* vehObj = ObjectManager::get<VehicleObject>(i);
+            if (vehObj == nullptr)
+            {
+                continue;
+            }
+
+            availableTypes |= 1ULL << enumValue(vehObj->type);
```

Comment:
```suggestion
            availableTypes |= 1U << enumValue(vehObj->type);
```
availableTypes here is only 16 bits